### PR TITLE
add http api to akka-sample-cqrs-scala

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+object ShoppingCartRoutes {
+  final case class AddItem(cartId: String, itemId: String, quantity: Int)
+  final case class UpdateItem(cartId: String, itemId: String, quantity: Int)
+  final case class Checkout(cartId: String)
+}
+
+class ShoppingCartRoutes()(implicit system: ActorSystem[_]) {
+
+  implicit private val timeout: Timeout = 5.seconds
```
Comment (johanandren) [inline akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala]: I'd put this in config as well when you anyway have system in scope

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        post {
+          entity(as[AddItem]) {
+            data =>
+              val entityRef = sharding.entityRefFor(ShoppingCart.EntityKey, data.cartId)
+              val reply: Future[ShoppingCart.Confirmation] =
+                entityRef.ask(ShoppingCart.AddItem(data.itemId, data.quantity, _))
+              onSuccess(reply) {
+                case ShoppingCart.Accepted(_) =>
+                  complete(
+                    StatusCodes.Accepted -> s"added item ${data.itemId}: ${data.quantity} to cart ${data.cartId}")
```
Comment (johanandren) [inline akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala]: A bit unconventional accepting Json but responding with text, ok for a sample I guess.

Comment (patriknw) [reply akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala]: changed to return the summary

File: akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala (hunk 1 above)
Comment (patriknw) [reply akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala]: done

**Comment by chbatey** [comment]
Failed was a different sample, created https://github.com/akka/akka-samples/issues/157 and re-running

**Pull request merged**